packages/dds/tree/src/simple-tree/core/treeNodeSchema.ts

```diff
@@ -94,7 +95,8 @@ export interface TreeNodeSchemaNonClass<
 export interface TreeNodeSchemaClass<
 	out Name extends string = string,
 	out Kind extends NodeKind = NodeKind,
-	out TNode = unknown,
+	// TODO: maybe this can be more specific (exclude leaves)
```

Comment:
That would amount to simply omitting `TreeLeafValue`, right? Is there a reason that doesn't work right now? Or, are we not actually sure that it would be a good thing to do?

Reply:
That makes this change even more breaking (restricts what types of schema can be used here) and complicates the definition of TreeNode schema since it has to become conditional since it can't pass the node type down to TreeNodeSchemaClass, and breaks isTreeNodeSchemaClass in a way I can't figure out how to fix (and that code which a user could write: breaking it in an unfixable way seems bad)